dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java

```diff
@@ -135,4 +138,25 @@ private void shutdown() {
         }
     }
 
+    private <K, V> KafkaConsumer<K, V> getKafkaConsumerWithRetry(Map<String, Object> consumerConfig, int retrySeconds, String channelName) throws InterruptedException {
+        if (retrySeconds == 0) {
+            return this.kafkaAdapterFactory.newKafkaConsumer(consumerConfig);
+        }
+
+        long retryNs = Duration.ofSeconds(retrySeconds).toNanos();
+        long startTime = System.nanoTime();
+
+        while (true) {
+            try {
+                return this.kafkaAdapterFactory.newKafkaConsumer(consumerConfig);
+            } catch (KafkaAdapterException e) {
+                if ((System.nanoTime() - startTime) > retryNs) {
+                    throw e;
+                }
+                Tr.warning(tc, "kafka.create.incoming.retry.CWMRX1009W", channelName, e.getMessage());
+            }
+            Thread.sleep(1000);
```

Comment:
Please extract this sleep time to a constant. Any thoughts on if it needs to be configurable?

Reply:
It's a fairly limited use case (broker hostname is not resolvable at startup) so I went for a simpler single option configuration, rather than two options (number of times to retry + delay between retries).

Reply:
I'm not sure that extracting it into a constant actually helps anything here.